Shared/Bytes.swift

```diff
@@ -11,8 +11,13 @@ public typealias GUID = String
  */
 open class Bytes {
     open class func generateRandomBytes(_ len: UInt) -> Data {
-        let bytes = [UInt32](repeating: 0, count: Int(len)).map { _ in arc4random() }
-        return Data(bytes: bytes, count: Int(len))
+        let len = Int(len)
+        var data = Data(count: len)
+        data.withUnsafeMutableBytes { (p: UnsafeMutablePointer<UInt8>) in
+            let result = SecRandomCopyBytes(kSecRandomDefault, len, p)
+            assert(result == 0, "Random byte generation failed.")
```

Comment:
I would be quite happy to have the client crash hard if `SRCB` returns anything other than `errSecSuccess`. We shouldn't proceed with whatever needed the value, and that means creating databases, computing GUIDs, or syncing. Crashing is our main way of finding out that there's a problem.

If we're building with normal optimizations, then switch `assert` for `precondition`.

If we're building with `-Ounchecked`, then you'll need to manually implement precondition using `fatalError`:

```swift
if (result != 0) {
    fatalError("Random byte generation failed.")
}
```